gensim/models/word2vec.py

```diff
@@ -327,9 +328,10 @@ class Word2Vec(utils.SaveLoad):
     compatible with the original word2vec implementation via `save_word2vec_format()` and `load_word2vec_format()`.
 
     """
-    def __init__(self, sentences=None, size=100, alpha=0.025, window=5, min_count=5,
-                 sample=0, seed=1, workers=1, min_alpha=0.0001, sg=1, hs=1, negative=0,
-                 cbow_mean=0, hashfxn=hash, iter=1, null_word=0):
+    def __init__(
+            self, sentences=None, size=100, alpha=0.025, window=5, min_count=5,
+            max_vocab_size=10000000, sample=0, seed=1, workers=1, min_alpha=0.0001,
```

Comment:
Unfortunately, any default means inaccuracy for those with plenty of memory – and a behavior change from prior versions they might not expect/notice. 

Yet, any default (10m-40m) still leaves problems (1GB-4GB) for those not thinking about memory at all, trying to do too much in 4/8/16GB. They'll hit an allocation error sooner or later, and need a guide to what they can try then.  I'd rather help them with better error messages/docs, than new defaults that power-users might not like. 

As a power-user, I'll always want to run with `max_vocab_size=0`, preferring to see a explicit failure if my vocabularies ever grow unexpectedly large – and then making a conscious adaptation, either in preprocessing or via `max_vocab_size`. So just having to add this new param to my instantiations to get the old behavior will be a minor annoyance.